lib/src/global_packages.dart

```diff
@@ -605,7 +527,9 @@ class GlobalPackages {
       var script = package.pubspec.executables[executable];
 
       var previousPackage = _createBinStub(package, executable, script,
-          overwrite: overwriteBinStubs, snapshot: snapshots[script]);
+          overwrite: overwriteBinStubs,
+          snapshot: entrypoint
+              .snapshotPathOfExecutable(Executable(package.name, executable)));
```

Comment:
Adding a trailing comma to the `_creatBinStub` args will make this format nicer 😄